src/api/buildPayTransaction.ts

```diff
@@ -1,37 +1,50 @@
-import { base } from 'viem/chains';
-import { PAY_HYDRATE_CHARGE } from '../network/definitions/pay';
-import { sendRequest } from '../network/request';
-import { PAY_UNSUPPORTED_CHAIN_ERROR_MESSAGE } from '../pay/constants';
+import {
+  CDP_CREATE_PRODUCT_CHARGE,
+  CDP_HYDRATE_CHARGE,
+} from '../network/definitions/pay';
+import { type JSONRPCResult, sendRequest } from '../network/request';
 import type {
   BuildPayTransactionParams,
   BuildPayTransactionResponse,
+  CreateProductChargeParams,
   HydrateChargeAPIParams,
 } from './types';
 import { getPayErrorMessage } from './utils/getPayErrorMessage';
 
 export async function buildPayTransaction({
   address,
-  chainId,
   chargeId,
+  productId,
 }: BuildPayTransactionParams): Promise<BuildPayTransactionResponse> {
-  if (chainId !== base.id) {
-    return {
-      code: 'AmBPTa01', // Api Module Build Pay Transaction Error 01
-      error: 'Pay Transactions must be on Base',
-      message: PAY_UNSUPPORTED_CHAIN_ERROR_MESSAGE,
-    };
-  }
   try {
-    const res = await sendRequest<
-      HydrateChargeAPIParams,
-      BuildPayTransactionResponse
-    >(PAY_HYDRATE_CHARGE, [
-      {
-        sender: address,
-        chainId: chainId,
-        chargeId,
-      },
-    ]);
+    let res: JSONRPCResult<BuildPayTransactionResponse>;
+    if (chargeId) {
```

Comment:
I'm not sure when we should expect chargeId and productId, maybe add a comment?

Reply:
they're two different methods and documented here - https://github.com/coinbase/onchainkit/pull/1338/files#diff-b7dd4bc7cc964af915e429bc56534f129be883d8ace8ace0523cc298b167965dR22-R24

Reply:
improved the comment here though, but I don't think we should duplicate docstrings across helper/util functions